db/blip_handler_collections.go

```diff
@@ -74,10 +74,9 @@ func (bh *blipHandler) handleGetCollections(rq *blip.Message) error {
 				checkpoints[i] = Body{}
 				bh.collectionMapping = append(bh.collectionMapping, collectionDB)
 			} else {
-				// TODO: CBG-2203 - should we return an error such that the client disconnects here?
-				base.WarnfCtx(bh.loggingCtx, "Unable to fetch client checkpoint %q for collection %s.%s", key, *scope, *collectionName)
-				checkpoints[i] = nil
-				bh.collectionMapping = append(bh.collectionMapping, nil)
+				errMsg := fmt.Sprintf("Unable to fetch client checkpoint %q for collection %s: %s", key, scopeAndCollection, err)
+				base.WarnfCtx(bh.loggingCtx, errMsg)
+				return base.HTTPErrorf(status, errMsg)
```

Comment:
Given the special handling of error codes on the Lite side - we probably need to change `status` to be a fixed `503` instead to give us the behaviour we want (the client reconnecting)
I don't know what clients would do if the error we returned to them was something else.